tools/AutoTuner/src/autotuner/utils/plot.py

```
@@ -0,0 +1,106 @@
+import glob
+import json
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import re
+import os
+import argparse
+
+AT_REGEX = r"variant-AutoTunerBase-([\w-]+)-\w+"
+IMG_DIR = "images"
+
+
+def load_dir(dir: str) -> pd.DataFrame:
+    # Concatenate progress DFs
+    df = pd.concat([pd.read_csv(fname) for fname in glob.glob(f"{dir}/*/progress.csv")])
+
+    # Concatenate params.json & metrics.json file
+    params = []
+    for fname in glob.glob(f"{dir}/*/params.json"):
+        try:
+            with open(fname, "r") as f:
+                _dict = json.load(f)
+                _dict["trial_id"] = re.search(AT_REGEX, fname).group(1)
+            with open(fname.replace("params.json", "metrics.json"), "r") as f:
+                metrics = json.load(f)
+                ws = metrics["finish"]["timing__setup__ws"]
+                metrics["worst_slack"] = ws
+                _dict.update(metrics)
+            params.append(_dict)
+        except Exception as e:
+            print(f"Error in {fname}: {e}")
+            continue
```

Should we keep a list of the files that were not processed so we can print them at the very end in order for the error not to be buried in the logs? And should the script fail silently in this case?

---

> And should the script fail silently in this case?
Just to clarify, this exception happens when any Autotuner run does not proceed to the "finish" stage, it does not fail silently

> Should we keep a list of the files that were not processed so we can print them at the very end in order for the error not to be buried in the logs? 
Sure I can add that.